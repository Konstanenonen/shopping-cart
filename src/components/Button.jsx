import React from 'react';
import PropTypes from 'prop-types';

function Button({ children, handleClick }) {
  return (
    <button onClick={handleClick} type="button">
      {children}
    </button>
  );
}

Button.defaultProps = {
  handleClick: null,
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  handleClick: PropTypes.func,
};

export default Button;
