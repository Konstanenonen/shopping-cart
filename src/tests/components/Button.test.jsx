/* global it describe expect jest */

import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import Button from '../../components/Button';

describe('Button component', () => {
  it("renders text 'CLICK ME'", () => {
    const { container } = render(<Button>CLICK ME</Button>);
    expect(container).toMatchSnapshot();
  });

  it('calls handleClick correct amount of times', () => {
    const handleClickMock = jest.fn();
    render(<Button handleClick={handleClickMock}>CLICK ME</Button>);
    const button = screen.getByRole('button', { name: 'CLICK ME' });

    act(() => {
      userEvent.click(button);
      userEvent.click(button);
      userEvent.click(button);
    });

    expect(handleClickMock).toBeCalledTimes(3);
  });
});
