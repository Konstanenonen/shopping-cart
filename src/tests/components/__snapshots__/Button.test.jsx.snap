// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders text 'CLICK ME' 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    CLICK ME
  </button>
</div>
`;
